src/components/form/__spec__.js

```diff
@@ -407,101 +411,100 @@ describe('Form', () => {
-        );
-        buttons = TestUtils.scryRenderedDOMComponentsWithTag(instance, 'button')
-        expect(buttons[0].disabled).toBeTruthy();
-      });
-
-      describe('when saveButtonProps is passed', () => {
-        it('sets save button props', () => {
-          let theme = 'magenta';
-
-          instance = TestUtils.renderIntoDocument(<Form saveButtonProps={ { theme: theme } } />);
-          buttons = TestUtils.scryRenderedDOMComponentsWithTag(instance, 'button')
-          expect(buttons[0].className).toContain(`carbon-button--${theme}`);
+      describe('the save button', () => {
+        it('by default it renders a save button', () => {
+          expect(wrapper.find('.carbon-form-save')).toBeTruthy();
         });
       });
 
-      describe('when cancelButtonProps is passed', () => {
-        it('sets cancel button props', () => {
-          let theme = 'red';
+      describe('the buttons', () => {
+        let wrapper;
+
+        describe('when no additional buttons are passed in', () => {
+          beforeEach(() => {
+            wrapper = shallow(
+              <Form
+                cancelText='Some custom text'
+                saveText='Some custom save text'
+                saving={ false }
+                saveButtonProps={ { theme: 'red' } }
+              />
+            )
+            wrapper.setState({ errorCount: 2, warningCount: 3 });
+          });
+
+          it('renders a cancel button with expected props', () => {
+            let cancelButton = wrapper.find(CancelButton);
+            expect(cancelButton.prop('cancelText')).toEqual('Some custom text')
+          });
+
+          it('renders a save button with expected props', () => {
+            let saveButton = wrapper.find(SaveButton);
+            expect(saveButton.prop('saveText')).toEqual('Some custom save text');
+            expect(saveButton.prop('saving')).toBeFalsy();
+            expect(saveButton.prop('saveButtonProps')).toEqual({ theme: 'red' })
+            expect(saveButton.prop('errors')).toEqual(2);
+            expect(saveButton.prop('warnings')).toEqual(3);
+          });
+        });
 
-          instance = TestUtils.renderIntoDocument(<Form cancelButtonProps={ { theme: theme } } />);
-          buttons = TestUtils.scryRenderedDOMComponentsWithTag(instance, 'button')
-          expect(buttons[1].className).toContain(`carbon-button--${theme}`);
+        describe('when an additional save button is passed in', () => {
+          beforeEach(() => {
+            let customButton = (<Button className='my-custom-class'>Save</Button>)
+            wrapper = mount(
+              <Form
+                cancelText='Some custom text'
+                saveText='Some custom save text'
+                saving={ false }
+                customSaveButton={ customButton }
+              />
+            )
+          });
+
+          it('does not render the standard SaveButton', () => {
+            expect(wrapper.find(SaveButton).length).toEqual(0);
```

Comment:
this can be written as ` expect(wrapper.find(SaveButton).exists()).toBeFalsy();`